### PR TITLE
Add ability to simulate x, y and z gradients

### PR DESCRIPTION
**Pull request by po09i**
## Description
This PR adds the ability to simulate x, y and z gradients

### CODE REVIEW

File: shimmingtoolbox/simulate/numerical_model.py [hunk 1]
```diff
@@ -107,30 +106,69 @@ def generate_deltaB0(self, field_type, params):
...
-                field (Hz).
+            field_type (str): Type of field to be generated. Available implementations are: ``'x'``, ``'y'``, ``'z'``.
+            params (list): List of parameters defining the field for the selected
+                           field type. If ``field_type = 'x' or 'y' or 'z'``, then ``params`` are
+                           ``[m b]`` where m (Hz/pixel) is the slope and b is the floor
+                           field (Hz).
         """
 
-        if field_type == "linear":
+        if field_type == "x":
```
Comment (4rnaudB) [inline shimmingtoolbox/simulate/numerical_model.py:131]: Would do:
if len(dims) == 2:
        [X, Y] = np.meshgrid(
         np.linspace(-dims[0], dims[0], dims[0]),
         np.linspace(-dims[1], dims[1], dims[1]))
        
if len(dims) == 3:
         [X, Y, Z] = np.meshgrid(
                        np.linspace(-dims[0], dims[0], dims[0]),
                        np.linspace(-dims[1], dims[1], dims[1]),
                        np.linspace(-dims[2], dims[2], dims[2]))

and then check for which gradient to set self.deltaB0 = m * Z + b

Comment (po09i) [reply shimmingtoolbox/simulate/numerical_model.py:131]: Good idea

File: shimmingtoolbox/simulate/numerical_model.py [hunk 2]
```diff
@@ -112,60 +112,30 @@ def generate_deltaB0(self, field_type, params):
...
-            if len(dims) == 2:
-                Z = 0
-
-            if len(dims) == 3:
-                [_, _, Z] = np.meshgrid(
-                    np.linspace(-dims[0], dims[0], dims[0]),
-                    np.linspace(-dims[1], dims[1], dims[1]),
-                    np.linspace(-dims[2], dims[2], dims[2]))
+        elif field_type == "z":
             self.deltaB0 = m * Z + b
```
Comment (4rnaudB) [inline shimmingtoolbox/simulate/numerical_model.py]: Do we want to return b if len(dim) == 2 and field_type =='z'?

Comment (po09i) [reply shimmingtoolbox/simulate/numerical_model.py]: Yes, this is currently the case because Z is getting set to 0 if it is 2D

**Review by 4rnaudB — APPROVED** [review]

**Pull request merged**